src/util/string.cpp

```diff
@@ -4,6 +4,7 @@
 
 #include <util/string.h>
 
+#include <map>
```

Comment:
This is wrong. Maybe use libc++ instead of libstdc++? Though, that will probably be wrong as well, see https://github.com/include-what-you-use/include-what-you-use/issues/1110#issuecomment-1245370908

Reply:
What about
```diff
--- a/src/util/string.cpp
+++ b/src/util/string.cpp
@@ -11,5 +11,5 @@
 void ReplaceAll(std::string& in_out, const std::string& search, const std::string& substitute)
 {
     if (search.empty()) return;
-    in_out = std::regex_replace(in_out, std::regex(search), substitute);
+    in_out = std::regex_replace(in_out, std::regex(search.data()), substitute);
 }
```
?

Looking into `bits/regex.h` makes me believe it won't introduce any regressions.

Reply:
There is a problem if we are refactoring code for no other reason than to make a buggy tool happy.

Reply:
https://github.com/bitcoin/bitcoin/blob/682a87e0db1fbddf20c40ab6bce9c70cad62e87a/src/util/string.cpp#L9-L10